### DISCUSSION TITLE
Percolation pathsways

### DISCUSSION THREAD

**Discussion by JahanzebTariq** [PMEAL/OpenPNM#2645] [Q&A]
Dear Professors @jgostick  and @ma-sadeghi. Please tell me **if there is a way in PNM to get the percolation pathways**. If so then please **guide me** or at least direct me to an **example**.
Thank you in advance

**Comment by jgostick** [comment]
When you run either invasion percolation or drainage, one of the arrays that get calculated is called 'pore.invasion_sequence'.  This tells you which order each pore was invaded in. With invasion percolation each pore (and throat) are numbered uniquely since the process occurs in individual steps, while in drainage (ordinary percolation), there are groups of pores and throats that are invaded at the same time.

**Reply by JahanzebTariq** [reply]
But **how can I visualize these pathways** ...  can you **please refer an example**

**Reply by jgostick** [reply]
Paraview?  Then you can color the throats by their invasion sequence, and even do a movie by changing the threshold value.  I'm sorry but we don't have an example of this.

**Reply by JahanzebTariq** [reply]
@jgostick .. sir I tried to do what you said but I came across this error

![image](https://user-images.githubusercontent.com/99289270/213394676-8bda97c1-7a0d-454e-8520-0ec16c172964.png)

please help me with this one.. thank you
